life_expectancy/i_o.py

```diff
@@ -1,21 +1,72 @@
+from abc import ABC, abstractmethod
+import zipfile
 import pandas as pd
+from life_expectancy.countries import Country
 
-def load_data(path):
+class FileReader(ABC):
+    """
+    Abstract class for reading different types of files
+    """
+    ext=''
+    @abstractmethod
+    def read_file(self, file_path):
+        """
+        Abstract method for reading different types of files
+        """
+        pass
+
+class TsvFileReader(FileReader):
+    """
+    Class to read tsv files that inherits from the FileReader abstract class
+    """
+    ext='tsv'
+    def read_file(self, file_path: str):
+        # code to read CSV file
+        df=pd.read_csv(file_path, sep='\t|,',engine ='python')
+        return df
+
+class ZippedJsonFileReader(FileReader):
+    """
+    Class to read zipped json files that inherits from the FileReader abstract class
+    """
+    ext='zip'
+    def read_file(self, file_path: str):
+        # code to read zipped JSON file
+        with zipfile.ZipFile(file_path) as zip_file:
+            json_file_name = None
+            for name in zip_file.namelist():
+                if name.endswith('.json'):
+                    json_file_name = name
```

Comment:
This bit is a bit weird... I see that you are trying to check for multiple json files, but this will overwrite `json_file_name` with a new value every time. You will essentially only load the last json file scanned.

Here's a suggestion

```
    with zipfile.ZipFile(zip_file_path, 'r') as zip_file:
        json_files = [f for f in zip_file.namelist() if f.endswith('.json')]
        data = [json.load(zip_file.open(json_file)) for json_file in json_files]
    return pd.concat([pd.DataFrame(d) for d in data], ignore_index=True)
```

Reply:
In this case as both versions solve what is intended, read one json file in a zip, I will keep my version because it adds interpretability, in my opinion, in comparison to yours with two list comprehensions and a concat.

Reply:
I assumed that what you wanted to do was to read all json files, and not a single one. But if you do want to read a single file, get can also remove that for look and if and do something much simpler.

```
def read_file(self, file_path: str) -> pd.DataFrame:
    """Load JSON data and returns a pandas DataFrame."""
    with zipfile.ZipFile(file_path) as zip_file:
        zipped_fname = zip_file.namelist()[0]
        with zip_file.open(zipped_fname) as json_file:
            df = pd.read_json(json_file)
    return df
```

This makes the function shorter, increases readability, and removes all loops and conditionals. Plus it raises a FileNotFound  error if there are no json files found (right now, it would raise a NameError with is less descriptive).

Reply:
Addresed in latest commit